examples/trusted_database/client/rust/src/main.rs

```diff
@@ -0,0 +1,120 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Client for the Trusted Database example.
+
+pub mod proto {
+    tonic::include_proto!("oak.examples.trusted_database");
+}
+
+use anyhow::{anyhow, Context};
+use log::info;
+use oak_abi::label::Label;
+use prost::Message;
+use proto::{
+    trusted_database_client::TrustedDatabaseClient, ListPointsOfInterestRequest, Location,
+};
+use structopt::StructOpt;
+use tonic::{
+    metadata::MetadataValue,
+    transport::{Certificate, Channel, ClientTlsConfig},
+    Request,
+};
+
+#[derive(StructOpt, Clone)]
+#[structopt(about = "Trusted Database Client")]
+pub struct Opt {
+    #[structopt(
+        long,
+        help = "URI of the Oak application to connect to",
+        default_value = "https://localhost:8080"
+    )]
+    uri: String,
+    #[structopt(
+        long,
+        help = "PEM encoded X.509 TLS root certificate file used by gRPC client"
+    )]
+    root_tls_certificate: String,
+    #[structopt(long, help = "Requested location's latitude in degrees (WGS84)")]
+    latitude: f32,
+    #[structopt(long, help = "Requested location's longitude in degrees (WGS84)")]
+    longitude: f32,
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    env_logger::init();
+    let opt = Opt::from_args();
+
+    let uri = opt.uri.parse().context("Error parsing URI")?;
+    let root_tls_certificate = tokio::fs::read(&opt.root_tls_certificate)
+        .await
+        .context("Couldn't load certificate file")?;
+    let latitude = opt.latitude;
+    if latitude < -90.0 || latitude > 90.0 {
+        return Err(anyhow!(
+            "Latitude must be a valid floating point number >=-90 and <= 90, found {}",
+            latitude
+        ));
+    }
+    let longitude = opt.longitude;
+    if longitude < -180.0 || longitude > 180.0 {
+        return Err(anyhow!(
+            "Longitude must be a valid floating point number >= -180 and <= 180, found {}",
+            longitude
+        ));
+    }
+
+    info!("Connecting to Oak Application: {:?}", uri);
+    let tls_config =
+        ClientTlsConfig::new().ca_certificate(Certificate::from_pem(root_tls_certificate));
+    let channel = Channel::builder(uri)
+        .tls_config(tls_config)
+        .context("Couldn't create TLS configuration")?
+        .connect()
+        .await
+        .context("Couldn't connect to Oak Application")?;
+
+    // TODO(#1097): Turn the following logic into a proper reusable client library.
+    let mut label = Vec::new();
+    Label::public_untrusted()
+        .encode(&mut label)
+        .context("Error encoding label")?;
+    let mut client =
+        TrustedDatabaseClient::with_interceptor(channel, move |mut request: Request<()>| {
+            request.metadata_mut().insert_bin(
```

Comment:
FWIW, I think eventually the labelling logic would benefit from being encapsulated in an interceptor anyways eventually, so this may actually be closer to what we expect in the final form. (similarly for authentication, which is not here)